pwndbg/commands/heap.py

```diff
@@ -214,33 +211,30 @@ def top_chunk(addr=None):
     main_heap   = pwndbg.heap.current
     main_arena  = main_heap.get_arena(addr)
 
-    if main_arena is None:
-        heap_region = main_heap.get_heap_boundaries()
-        if not heap_region:
-            print(message.error('Could not find the heap'))
-            return
-
-        heap_start = heap_region.vaddr
-        heap_end   = heap_start + heap_region.size
-
-        # If we don't know where the main_arena struct is, just iterate
-        # through all the heap objects until we hit the last one
-        last_addr = None
-        addr = heap_start
-        while addr < heap_end:
-            chunk = read_chunk(addr)
-            size = int(chunk['size'])
-
-            # Clear the bottom 3 bits
-            size &= ~7
-
-            last_addr = addr
-            addr += size
-            addr += size
-        address = last_addr
-    else:
-        address = main_arena['top']
+    heap_region = main_heap.get_heap_boundaries()
+	
+    if not heap_region:
```

Comment:
It seems that we do the same logic in two places; can we move it to a function?

Reply:
I mean, the part below that if...

Reply:
sure

Reply:
> It seems that we do the same logic in two places; can we move it to a function?

you can check the new commit for this